### PR TITLE
[SMALLFIX] adding a parameter to support more flexible locking of children

### PR DESCRIPTION
**Pull request by yuzhu**

### CODE REVIEW

File: core/server/master/src/main/java/alluxio/master/file/meta/LockedInodePath.java [hunk 1]
```diff
@@ -231,15 +231,17 @@ public synchronized LockedInodePath createTempPathForChild(String childName)
...
    *
    * @param child the inode of the direct child
+   * @param lockMode the desired locking mode for the child
    * @return a {@link LockedInodePath} for the direct child
    * @throws InvalidPathException if the path is invalid
    * @throws FileDoesNotExistException if the file does not exist
    */
-  public synchronized LockedInodePath createTempPathForExistingChild(Inode<?> child)
+  public synchronized LockedInodePath createTempPathForExistingChild(Inode<?> child,
+                                                                     InodeTree.LockMode lockMode)
```
Comment (apc999) [inline core/server/master/src/main/java/alluxio/master/file/meta/LockedInodePath.java]: wrong indent

**Comment by apc999** [comment]
Inconsistent synchronization of alluxio.master.file.meta.LockedInodePath.mLockMode; locked 75% of time [alluxio.master.file.meta.InodeTree, alluxio.master.file.meta.LockedInodePath, alluxio.master.file.meta.LockedInodePath, alluxio.master.file.meta.LockedInodePath] Unsynchronized access at InodeTree.java:[line 830]Synchronized access at LockedInodePath.java:[line 223]Synchronized access at LockedInodePath.java:[line 190]Synchronized access at LockedInodePath.java:[line 163]

**Comment by yuzhu** [comment]
jenkins, test this please

**Comment by yuzhu** [comment]
comments addressed, build fixed. PTAL.

**Pull request merged**